packages/runtime/container-runtime/src/pendingStateManager.ts

```diff
@@ -328,15 +338,18 @@ export class PendingStateManager implements IDisposable {
 	 * @returns whether the batch IDs match
 	 */
 	private remoteBatchMatchesPendingBatch(remoteBatchStart: BatchStartInfo): boolean {
-		// We may have no pending changes - if so, no match, no problem.
-		const pendingMessage = this.pendingMessages.peekFront();
-		if (pendingMessage === undefined) {
+		// Find the first pending message that uses Batch ID, to compare to the incoming remote batch.
+		// If there is no such message, then the incoming remote batch doesn't have a match here and we can return.
+		const pendingMessageUsingBatchId = this.pendingMessages
```

Comment:
Can we avoid an allocation here? Doesn't Deque allow random access? I'm wondering if we need to pay an O(messages) cost here (and I don't know the flow well enough to answer whether or not we care).

Reply:
Isn't the case that idAllocation batch would be the first one being flushed? therefore we just need to look at the beginning of the list?

Reply:
I was worried about that too, @taylorsw04.  @dannimad - we don't know whether ID Allocation batch is the first one or not (e.g. if the feature is disabled...).

Ok I looked at the Deque API again, I can do a loop with `get` and it'll be faster.  I was wishing it implemented Iterable or whatever so I could do something a little more functional but this will be ok.